src/csvwidget/widget.ts

```diff
@@ -217,6 +217,27 @@ function createDelimiterSwitcherNode(): HTMLElement {
 export
 class CSVWidgetFactory extends ABCWidgetFactory<CSVWidget, DocumentRegistry.IModel> {
   /**
+   * The name of the widget to display in dialogs.
+   */
+  get name(): string {
+    return 'Table';
+  }
+
+  /**
+   * The file extensions the widget can view.
+   */
+  get fileExtensions(): string[] {
+    return ['.csv'];
```

Comment:
+1 to being able to override them, but sure, it makes sense to have some sensible defaults.